receiver/windowsperfcountersreceiver/config.go

```diff
@@ -22,20 +22,24 @@ import (
 	"go.opentelemetry.io/collector/receiver/receiverhelper"
 )
 
-// Config defines configuration for HostMetrics receiver.
+// Config defines configuration for WindowsPerfCounters receiver.
 type Config struct {
 	configmodels.ReceiverSettings            `mapstructure:",squash"`
 	receiverhelper.ScraperControllerSettings `mapstructure:",squash"`
 
 	PerfCounters []PerfCounterConfig `mapstructure:"perfcounters"`
 }
 
+// PerfCounterConfig defines configuration for a perf counter object.
 type PerfCounterConfig struct {
-	Object   string   `mapstructure:"object"`
-	Counters []string `mapstructure:"counters"`
+	Object    string   `mapstructure:"object"`
+	Instances []string `mapstructure:"instances"`
```

Comment:
Do you think we should consider being able to collect certain counters from only certain instances in the config? Or would one do this by using different receiver configs? For example, a case where one is interested in collecting counterA and counterB from instanceA only, and counterA and counterC from instanceB only.

Reply:
If I understand you correctly, this is supported, as per the example [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1229/files#diff-89063f3eceb46041b3b8d2a7ac2a936a99c35d3205d411b5ad5cbf4b9eaae1ceR62-R69) and test [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1229/files#diff-6fca9988c28bc88c215f08f9df2354795b7746050fbf3e319cf4ca146e5c856fR89-R90).

You can achieve what you want as there's no limitation that you can only specify an `Object` once.

Aside: there's currently no validation to stop you setting up completely duplicate counters, although I did leave a comment [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1229/files#diff-cc24dbb6212ea781b462531887122e0ffa0ffe5c913a48664aab5bad1ff60ec1R41) mentioning we should consider adding such validation in the future.

Reply:
Ah yes, that's what I was referring to, thanks for pointing me to it!